lib/src/boilerplate_suggestors/props_mixins_migrator.dart

```diff
@@ -16,12 +16,78 @@ import 'package:analyzer/dart/ast/ast.dart';
 import 'package:analyzer/dart/ast/visitor.dart';
 import 'package:codemod/codemod.dart';
 
+import 'boilerplate_utilities.dart';
+
 /// Suggestor that updates stand alone props mixins to be actual mixins.
+///
+/// > Related: [PropsMetaMigrator]
 class PropsMixinMigrator extends GeneralizingAstVisitor
     with AstVisitingSuggestorMixin
     implements Suggestor {
   @override
   visitClassDeclaration(ClassDeclaration node) {
     super.visitClassDeclaration(node);
+
+    if (!shouldMigratePropsAndStateMixin(node)) return;
+
+    migrateClassToMixin(node, yieldPatch);
+    _removePropsOrStateGetter(node);
+    _removePropsOrStateMixinAnnotation(node);
+    _migrateMixinMetaField(node);
+  }
+
+  void _removePropsOrStateGetter(ClassDeclaration node) {
+    final classGetters = node.members
+        .whereType<MethodDeclaration>()
+        .where((method) => method.isGetter);
+    final propsOrStateGetter = classGetters.singleWhere(
+        (getter) => getter.name.name == 'props' || getter.name.name == 'state',
+        orElse: () => null);
+    if (propsOrStateGetter != null) {
+      yieldPatch(propsOrStateGetter.offset, propsOrStateGetter.end + 1, '');
+    }
+  }
+
+  void _removePropsOrStateMixinAnnotation(ClassDeclaration node) {
+    final propsMixinAnnotationNode = getPropsMixinAnnotationNode(node);
+    if (propsMixinAnnotationNode != null) {
+      yieldPatch(propsMixinAnnotationNode.offset,
+          propsMixinAnnotationNode.end + 1, '');
+    }
+
+    final stateMixinAnnotationNode = getStateMixinAnnotationNode(node);
+    if (stateMixinAnnotationNode != null) {
+      yieldPatch(stateMixinAnnotationNode.offset,
+          stateMixinAnnotationNode.end + 1, '');
+    }
+  }
+
+  /// NOTE: Usage of the meta field elsewhere will be migrated via the [PropsMetaMigrator].
+  void _migrateMixinMetaField(ClassDeclaration node) {
+    final classMembers = node.members;
+    final classFields = classMembers
+        .whereType<FieldDeclaration>()
+        .map((decl) => decl.fields)
+        .toList();
+    final metaField = classFields.singleWhere(
+        (field) => field.variables.single.name.name == 'meta',
+        orElse: () => null);
+    if (metaField == null) return;
+
+    if (isPublic(node)) {
+      yieldPatch(metaField.parent.offset, metaField.parent.offset,
+          '@Deprecated(\'Use `propsMeta.forMixin(${node.name.name})` instead.\')\n');
+    } else {
+      // Remove the meta field, along with any comment lines that preceded it.
```

Comment:
This can be done using the `.precedingComments` field (see doc comment for more details).

```dart
final begin = metaField.precedingComments?.offset ?? metaField.offset;
```